### PR TITLE
View Pin feature. Bug fixes

### PR DESCRIPTION
**Pull request by gabriel-cer**

### CODE REVIEW

File: CardManagementSDK/Source/Network/ResponseModels/ViewPinResponse.swift [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+            print("Key '\(key)' not found:", context.debugDescription)
+            print("codingPath:", context.codingPath)
+        } catch DecodingError.valueNotFound(let value, let context) {
+            print("Value '\(value)' not found:", context.debugDescription)
+            print("codingPath:", context.codingPath)
+        } catch DecodingError.typeMismatch(let type, let context) {
+            print("Type '\(type)' mismatch:", context.debugDescription)
+            print("codingPath:", context.codingPath)
+        } catch {
+//            print("error: ", error)
```
Comment (philatov) [inline CardManagementSDK/Source/Network/ResponseModels/ViewPinResponse.swift]: Why?

File: CardManagementSDK/Source/Views/ViewPin/NIViewPinView.swift [hunk 2]
```diff
@@ -0,0 +1,233 @@
...
+    
+    // MARK: - Private
+
+    private func startTimer() {
+        timer = Timer.scheduledTimer(timeInterval: 1.0, target: self, selector: #selector(self.updateCounter), userInfo: nil, repeats: true)
+    }
+    
+    @objc private func updateCounter() {
+        if counter > 0 {
+            countDownDecription.text = "view_pin_countdown_description".localized + " " + String(Int(counter)) + " " + "view_pin_countdown_unit".localized
```
Comment (philatov) [inline CardManagementSDK/Source/Views/ViewPin/NIViewPinView.swift:95]: Are you sure that string concatenation will be the same for RTL?

File: CardManagementSDK/Source/Views/ViewPin/NIViewPinView.swift [hunk 3]
```diff
@@ -0,0 +1,233 @@
...
+            counter -= 1
+        } else {
+            timer?.invalidate()
+            maskPIN()
+        }
+    }
+    
+    private func maskPIN() {
+        firstDigit.text = "＊"
```
Comment (philatov) [inline CardManagementSDK/Source/Views/ViewPin/NIViewPinView.swift]: Can't we have a list of controls and access them via index, not names?

File: CardManagementSDK/Source/Views/ViewPin/NIViewPinView.swift [hunk 4]
```diff
@@ -0,0 +1,233 @@
...
+                    self.updateFirst4Digits(viewModel: viewModel)
+                    self.fifthDigit.text = viewModel.pin?[4]
+                    self.sixthDigit.text = viewModel.pin?[5]
+                }
+                
+                if viewModel.startTimer && self.counter != 0 {
+                    self.startTimer()
+                    let countDownDescriptionLocalized: String = "view_pin_countdown_description".localized + " "
+                    let countDownUnitLocalized: String = " " + "view_pin_countdown_unit".localized
+                    self.countDownDecription.text = countDownDescriptionLocalized + String(Int(self.counter)) + countDownUnitLocalized
```
Comment (philatov) [inline CardManagementSDK/Source/Views/ViewPin/NIViewPinView.swift:154]: Clarify that same concatenation is applicable for RTL.

File: CardManagementSDK/Source/Network/ResponseModels/ViewPinResponse.swift (hunk 1 above)
Comment (gabriel-cer) [reply CardManagementSDK/Source/Network/ResponseModels/ViewPinResponse.swift]: Screenshot 2023-11-01 at 14 29 11
Screenshot 2023-11-01 at 14 29 24
Did some tests and it looks good with phone in arabic (RTL)

File: CardManagementSDK/Source/Views/ViewPin/NIViewPinView.swift (hunk 4 above)
Comment (gabriel-cer) [reply CardManagementSDK/Source/Views/ViewPin/NIViewPinView.swift:154]: Looks fine in tests.

File: CardManagementSDK/Source/Views/ViewPin/NIViewPinView.swift (hunk 2 above)
Comment (gabriel-cer) [reply CardManagementSDK/Source/Views/ViewPin/NIViewPinView.swift:95]: Screenshot 2023-11-01 at 14 29 11
Screenshot 2023-11-01 at 14 29 24

Did some tests and it looks good with phone in arabic (RTL)

File: CardManagementSDK/Source/Network/ResponseModels/ViewPinResponse.swift (hunk 1 above)
Comment (gabriel-cer) [reply CardManagementSDK/Source/Network/ResponseModels/ViewPinResponse.swift]: Fixed.

File: CardManagementSDK/Source/Views/ViewPin/NIViewPinView.swift (hunk 3 above)
Comment (gabriel-cer) [reply CardManagementSDK/Source/Views/ViewPin/NIViewPinView.swift]: Fixed.

**Review by philatov — APPROVED** [review]

**Pull request merged**